### DISCUSSION TITLE
Add visual regression testing with Playwright Snapshots?

### DISCUSSION THREAD

**Discussion by Eric-Arellano** [pradyunsg/furo#661] [Ideas] [answered]
Hello. Would you be interested in me opening a PR to add visual regression testing? tl;dr is that you can use Microsoft's Playwright test runner to take before-and-after screenshots of certain parts of the site, such as the footer.

If the change was intentional, then you regenerate the snapshots. Otherwise, you have to fix your change to not regress.

See https://github.com/Qiskit/qiskit_sphinx_theme/pull/345 for more details and an example of what this PR might look like.

Imo, the main downside is the user needs to have Docker installed and running to test the snapshots so that local and CI are consistent.

**Comment by pradyunsg** [comment, marked as answer]
Yes, please do! No promises that I'll merge it as-is, but I'd love to have PR to either just merge or build off of! :)

**Reply by pradyunsg** [reply]
#662 was filed.
